src/mmw/apps/modeling/migrations/0023_fix_gis_data_serialization.py

```diff
@@ -9,10 +9,8 @@
 
 def fix_gis_data_serialization(apps, schema_editor):
     Project = apps.get_model('modeling', 'Project')
-    for project in Project.objects.all():
-        if project.gis_data:
-            # literal_eval works for both JSON and str(dict)
-            # https://docs.python.org/2.7/library/ast.html#ast.literal_eval
+    for project in Project.objects.exclude(gis_data=''):
+        if "u'" in project.gis_data:
```

Comment:
This `if` can probably be included in the filter like so:

```python
for project in Project.objects.filter(gis_data__startswith='{u'):
    project.gis_data = json.dumps(ast.literal_eval(project.gis_data))
    project.save()
```

See [startswith](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#std:fieldlookup-startswith)

Reply:
Thanks. I pushed another commit with this more efficient query.

The count of matching projects on staging

```
>>> Project.objects.filter(gis_data__startswith='{u').count()
14
```

Matches the count from this query on staging

```
SELECT _left, COUNT(_left)
FROM (
    SELECT LEFT(gis_data, 10) AS _left
    FROM modeling_project
    WHERE gis_data IS NOT NULL) AS x
GROUP BY _left;

   _left    | count
------------+-------
 {"n38": 0, |    80
 {u'n38': 0 |    14
(2 rows)
```